### PR TITLE
Analysis cleanup: simplify redundant if statement

### PR DESCRIPTION
**Pull request by DanVanAtta**
Return boolean condition directly instead of (eg) doing an if clause around a 'return true'

### CODE REVIEW

File: src/main/java/games/strategy/engine/data/properties/NumberProperty.java [hunk 1]
```diff
@@ -63,9 +63,7 @@ public JComponent getEditorComponent() {
   public boolean validate(final Object value) {
     if (value instanceof Integer) {
       final int i = ((Integer) value);
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/engine/data/properties/NumberProperty.java]: Not directly related, but the outer parentheses can be removed, and the value can be cast to int directly

File: src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java [hunk 2]
```diff
@@ -102,9 +102,7 @@ private boolean testShouldWeAbort() {
     }
 
     if (serverGame != null && serverGame.getPlayerManager() != null) {
-      if (serverGame.getPlayerManager().isEmpty()) {
-        return true;
-      }
+      return serverGame.getPlayerManager().isEmpty();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java]: Not directly related either, but now the if statement can be merged with both return statements, by combining both boolean expressions with a logical AND

File: src/main/java/games/strategy/net/IPFinder.java [hunk 3]
```diff
@@ -93,16 +93,11 @@ private static boolean isPrivateNetworkAddress(final InetAddress address) {
...
+      return (bytes[0] == 10) || (bytes[0] == octet172 && bytes[1] >= 16 && bytes[1] <= 31)
+          || (bytes[0] == octet192 && bytes[1] == octet168) || (bytes[0] == octet169 && bytes[1] == octet254);
     } else { // ip 6
       // http://en.wikipedia.org/wiki/IPv6#Addressing
-      if ((bytes[0] == octet252 && bytes[1] == 0) || bytes[0] == octet254) {
-        return true;
-      }
+      return (bytes[0] == octet252 && bytes[1] == 0) || bytes[0] == octet254;
     }
-    return false;
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/net/IPFinder.java]: The explicit else block now becomes redundant.

File: src/main/java/games/strategy/triplea/attachments/TerritoryAttachment.java [hunk 4]
```diff
@@ -33,15 +33,10 @@ public static boolean doWeHaveEnoughCapitalsToProduce(final PlayerID player, fin
...
     if (pa == null) {
-      if (!capitalsListOriginal.isEmpty() && capitalsListOwned.isEmpty()) {
-        return false;
-      }
+      return capitalsListOriginal.isEmpty() || !capitalsListOwned.isEmpty();
     } else {
-      if (pa.getRetainCapitalProduceNumber() > capitalsListOwned.size()) {
-        return false;
-      }
+      return pa.getRetainCapitalProduceNumber() <= capitalsListOwned.size();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/attachments/TerritoryAttachment.java]: The explicit else becomes redundant here as well.

File: src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java [hunk 5]
```diff
@@ -1623,19 +1623,15 @@ protected boolean isUnitPlacementRestrictions() {
   protected boolean isPlayerAllowedToPlacementAnyTerritoryOwnedLand(final PlayerID player) {
     if (isPlaceInAnyTerritory()) {
       final RulesAttachment ra = (RulesAttachment) player.getAttachment(Constants.RULES_ATTACHMENT_NAME);
-      if (ra != null && ra.getPlacementAnyTerritory()) {
-        return true;
-      }
+      return ra != null && ra.getPlacementAnyTerritory();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java]: Not sure if this works, but perhaps this statement can be combined with the Boolean expression in the wrapping if clause if the rules attachment doesn't depend on the result of this expression.

File: src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java [hunk 6]
```diff
@@ -1623,19 +1623,15 @@ protected boolean isUnitPlacementRestrictions() {
...
     return false;
   }
 
   protected boolean isPlayerAllowedToPlacementAnySeaZoneByOwnedLand(final PlayerID player) {
     if (isPlaceInAnyTerritory()) {
       final RulesAttachment ra = (RulesAttachment) player.getAttachment(Constants.RULES_ATTACHMENT_NAME);
-      if (ra != null && ra.getPlacementAnySeaZone()) {
-        return true;
-      }
+      return ra != null && ra.getPlacementAnySeaZone();
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java]: See above

File: src/main/java/games/strategy/triplea/delegate/Matches.java [hunk 7]
```diff
@@ -1490,10 +1483,8 @@ private Matches() {}
         final Collection<Unit> airTransports = CollectionUtils.getMatches(units, unitIsAirTransport());
         final Collection<Unit> paratroops = CollectionUtils.getMatches(units, unitIsAirTransportable());
         if (!airTransports.isEmpty() && !paratroops.isEmpty()) {
-          if (TransportUtils.mapTransportsToLoad(paratroops, airTransports)
-              .containsKey(dependent)) {
-            return true;
-          }
+          return TransportUtils.mapTransportsToLoad(paratroops, airTransports)
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/Matches.java]: This can be merged with the surrounding if statement as well

**Comment by DanVanAtta** [comment]
Good suggestions @RoiEXLab , but it's a step further than I want for this branch. I'll take care of the redundant else blocks, the rest I suspect I'll go down a rabbit hole to fix each one, and would probably belong on their own branches. This PR is mostly just applying some static fixes with quick cleanup, the redundant else blocks I can loop into that happily enough.

**Comment by DanVanAtta** [comment]
f91338cbe will take care of the low hanging fruit.

**Comment by DanVanAtta** [comment]
e5da28e5c is following the rabbit hole a bit yet further. I think that is about as far as I'd like to increase the scope of this branch. My concern is not really that the cleanups should not be fixed, but that there are those and many more that also deserve to be fixed (if i start pulling on that string, I suspect it would go pretty far, too far).

**Pull request merged**